torch/_dynamo/variables/builder.py

```diff
@@ -634,6 +627,26 @@ def index_source(key):
                 source=self.source,
                 guards=self.make_guards(GuardBuilder.ID_MATCH),
             )
+        elif DeviceMeshVariable.is_device_mesh(value):
+            return DeviceMeshVariable(
+                value,
+                source=self.source,
+                guards=self.make_guards(GuardBuilder.ID_MATCH),
+            )
+        elif PlacementClassVariable.is_placement_type(value):
+            return PlacementClassVariable(
+                value,
+                source=self.source,
+                guards=make_guards(GuardBuilder.FUNCTION_MATCH),
```

Comment:
This is just ID_MATCH with a local scope. Use ID MATCH, or, better yet, write a custom guard for this type for the things you care about (Can be a TODO).